File: app/src/main/java/dev/ragnarok/fenrir/model/selection/LocalPhotosSelectableSource.java
package dev.ragnarok.fenrir.model.selection;

import android.os.Parcel;


public class LocalPhotosSelectableSource extends AbsSelectableSource {

    public static final Creator<LocalPhotosSelectableSource> CREATOR = new Creator<LocalPhotosSelectableSource>() {
        @Override
        public LocalPhotosSelectableSource createFromParcel(Parcel in) {
            return new LocalPhotosSelectableSource(in);
        }

        @Override
        public LocalPhotosSelectableSource[] newArray(int size) {
            return new LocalPhotosSelectableSource[size];
        }
    };

    public LocalPhotosSelectableSource() {
        super(Types.LOCAL_PHOTOS);
    }

    protected LocalPhotosSelectableSource(Parcel in) {
        super(in);
    }

    @Override
    public int describeContents() {
        return 0;
    }
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/model/selection/LocalPhotosSelectableSource.kt
package dev.ragnarok.fenrir.model.selection

import android.os.Parcel
import android.os.Parcelable

class LocalPhotosSelectableSource : AbsSelectableSource {
    constructor() : super(Types.LOCAL_PHOTOS)
    private constructor(`in`: Parcel) : super(`in`)

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<LocalPhotosSelectableSource> {
        override fun createFromParcel(parcel: Parcel): LocalPhotosSelectableSource {
            return LocalPhotosSelectableSource(parcel)
        }

        override fun newArray(size: Int): Array<LocalPhotosSelectableSource?> {
            return arrayOfNulls(size)
        }
    }
}